processor/lib/azureStorageCheckpointLeaseManager.ts

```diff
@@ -64,7 +64,8 @@ export class AzureStorageCheckpointLeaseManager implements CheckpointManager, Le
       const jsonLease: LeaseInfo = JSON.parse(text);
       const blobLeaseInfo: AzureBlobLeaseInfo = {
         ...jsonLease,
```

Comment:
`...jsonLease` should have the partitionId in it. The problem lies somewhere else, whenever the content was written to blob storage, it did not have partitionId in it. Can you provide some more logs, to help find out that root cause?

Reply:
Thanks for the feedback. I should mention again, the messages being sent to the event hub are coming from a C# EventHubClient. I followed the tutorials on the MSDN for setting it up, which can be found here.
- https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-programming-guide
- https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-event-processor-host

I looked on storage and found the following info in the partition blobs. It appears some are being written without partition ids.
```
$Default/0 - {"owner":"js-host-0826076f-42f3-4129-b1fb-6695732d532d","epoch":21,"sequenceNumber":0,"token":"85ed91f4-7223-49cf-ad20-b34b69cb5b06"}

$Default/1 - {"owner":"js-host-0826076f-42f3-4129-b1fb-6695733d532d","epoch":21,"sequenceNumber":0,"token":"cacab3c9-2d4a-475e-a622-f41baed48dbb"}

$Default/2 - {"Offset":null,"SequenceNumber":0,"PartitionId":"2","Owner":"host-09fa2b37-239f-4820-b1a3-6718ed9f6142","Token":"f4abca8d-233d-41df-ac8e-15df470b9ded","Epoch":21}

$Default/3 - {"owner":"js-host-0826076f-42f3-4129-b1fb-6695733d532d","epoch":22,"sequenceNumber":0,"token":"174f6f89-e253-4fd2-ae6d-210aad163c41"}
```

I suspect that partition 2 was generated from the C# code, while the other partitions were generated by the JS code. It seems like there is different casing and the host is a dead giveaway. On a side note, is this casing problematic? Possibly I shouldn't be mixing and matching C# and JS event hub client / processors?

Finally, here's the log of the start up with the failure.
```
azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Resetting the partition manager. +0ms
  azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Get the list of partition ids. +2ms
  azure:eph:util [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Retry attempt: 0. Action 'Getting PartitionIds' suceeded. +0ms
  azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Ensuring that the lease store exists. +1s
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Does container 'spew' exist -> true. +0ms
  azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Ensure the checkpointstore exists. +229ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Checking whether the checkpoint store exists. +0ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Does container 'spew' exist -> true. +54ms
  azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Ensure that the leases exist. +53ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Attempting to list blobs for container 'spew'. +2ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Successfully, received the list of blobs for container 'spew'. +115ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Number of blobs: 4 +171ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Number of blobs 4 === Number of partitionIds 4. Hence no need to create leases. +1ms
  azure:eph:util [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Retry attempt: 0. Action 'Creating Lease' suceeded. +403ms
  azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Ensure that the checkpoint exists. +120ms
  azure:eph:util [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Retry attempt: 0. Action 'Creating Checkpoint' suceeded. +2ms
  azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Starting the first scan. +3ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Starting a new iteration to scan partitions. +0ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Getting lease states to find out who owns what. +0ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Attempting to list blobs for container 'spew'. +9ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Successfully, received the list of blobs for container 'spew'. +104ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Number of blobs: 4 +113ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [0] BlobResult item from the list of blobs is: name: $Default/0, lease: { status: 'unlocked', state: 'expired' }, metadata: { owninghost: 'js-host-2c03fa0c-7fe9-4025-ab58-3727b60700f1' }. +4ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [1] BlobResult item from the list of blobs is: name: $Default/1, lease: { status: 'unlocked', state: 'expired' }, metadata: { owninghost: 'js-host-05277fe2-0422-4f81-8180-c4ba30a9ca12' }. +5ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] BlobResult item from the list of blobs is: name: $Default/2, lease: { status: 'unlocked', state: 'expired' }, metadata: { owninghost: 'js-host-2c03fa0c-7fe9-4025-ab58-3727b60700f1' }. +4ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [3] BlobResult item from the list of blobs is: name: $Default/3, lease: { status: 'unlocked', state: 'expired' }, metadata: { owninghost: 'js-host-2c03fa0c-7fe9-4025-ab58-3727b60700f1' }. +6ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Number of leases: 4 +5ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] State of all the leases that we know about:
  azure:eph:partitionScanner [ BaseLease {
  azure:eph:partitionScanner     isOwned: false,
  azure:eph:partitionScanner     partitionId: '0',
  azure:eph:partitionScanner     owner: 'js-host-2c03fa0c-7fe9-4025-ab58-3727b60700f1' },
  azure:eph:partitionScanner   BaseLease {
  azure:eph:partitionScanner     isOwned: false,
  azure:eph:partitionScanner     partitionId: '1',
  azure:eph:partitionScanner     owner: 'js-host-05277fe2-0422-4f81-8180-c4ba30a9ca12' },
  azure:eph:partitionScanner   BaseLease {
  azure:eph:partitionScanner     isOwned: false,
  azure:eph:partitionScanner     partitionId: '2',
  azure:eph:partitionScanner     owner: 'js-host-2c03fa0c-7fe9-4025-ab58-3727b60700f1' },
  azure:eph:partitionScanner   BaseLease {
  azure:eph:partitionScanner     isOwned: false,
  azure:eph:partitionScanner     partitionId: '3',
  azure:eph:partitionScanner     owner: 'js-host-2c03fa0c-7fe9-4025-ab58-3727b60700f1' } ] +134ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Sorting leases to find out the desired partition count. +22ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Accounting input: allLeaseStates count is: 4 +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Host ordinal: -1. Rotating leases to start at: 2. +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Host count is 1. So Desired partition count is 1. +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] our leases count: 0 v/s leases owned by others: 0. +0ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Total number of pumps: 0. +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Our lease count: 0. +2ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Attempting to acquire 1 leases in parallel starting from position 0. +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Examining chunk at '2': [0], needed 1. +2ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Finding expired leases from inclusive position range 0 - 1 +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Finding expired leases from '2': [0] upto '3': [1] +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Found in range: 1 +1ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Getting lease. +37ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Does blob '$Default/2' exist in container 'spew' -> true. +110ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Attempting to getcontent from blobPath '$Default/2'. +5ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Successfully, fetched blob content '{"owner":"js-host-2c03fa0c-7fe9-4025-ab58-3727b60700f1","epoch":5,"sequenceNumber":0,"token":"92b007f5-666b-4912-ac8a-94672222edd1"}' for blobPath '$Default/2'. +88ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Attempting to get blob props for blobPath '$Default/2'. +6ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Successfully, got blob props for blobPath '$Default/2'. +47ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Attempting to acquire lease for blobPath '$Default/2'. +2ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Acquired lease '0fc31599-8049-4b43-bf85-c78dca2228c6' for blobPath '$Default/2. +50ms
  azure:eph:completeLease [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [undefined] New epoch for lease is 6. +0ms
  azure:eph:completeLease [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [undefined] Lease info is: { partitionId: undefined, owner: 'js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2', epoch: 6 } +2ms
  azure:eph:azurebloblease [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [undefined] Lease info is: { partitionId: undefined, owner: 'js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2', epoch: 6, sequenceNumber: 0, token: '0fc31599-8049-4b43-bf85-c78dca2228c6', offset: undefined } +0ms
  azure:eph:checkpointLeaseMgr [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [undefined] Trying to upload raw JSON for activity 'acquire': {"owner":"js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2","epoch":6,"sequenceNumber":0,"token":"0fc31599-8049-4b43-bf85-c78dca2228c6"}, with options: { leaseId: '0fc31599-8049-4b43-bf85-c78dca2228c6', metadata: { owninghost: 'js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2' } } +257ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Updating content '{"owner":"js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2","epoch":6,"sequenceNumber":0,"token":"0fc31599-8049-4b43-bf85-c78dca2228c6"}' in the container 'spew' of the blob '$Default/2' . +17ms
  azure:eph:blobService [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Successfully, updated blob content '{"owner":"js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2","epoch":6,"sequenceNumber":0,"token":"0fc31599-8049-4b43-bf85-c78dca2228c6"}' for blobPath '$Default/2'. +66ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [2] Acquired unowned/expired lease. +333ms
  azure:eph:completeLease [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [undefined] Lease info is: { partitionId: undefined, owner: 'js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2', epoch: 6 } +85ms
  azure:eph:azurebloblease [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [undefined] Lease info is: { partitionId: undefined, owner: 'js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2', epoch: 6, sequenceNumber: 0, token: '0fc31599-8049-4b43-bf85-c78dca2228c6', offset: undefined } +86ms
  azure:eph:pumpManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [N/A] Creating a new pump with lease { partitionId: undefined, owner: 'js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2', epoch: 6, sequenceNumber: 0, token: '0fc31599-8049-4b43-bf85-c78dca2228c6', offset: undefined }. +0ms
  azure:eph:partitionPump [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [N/A] Getting the initial offset. +0ms
  azure:eph:error [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] [N/A] An error occurred while adding/updating a pump for partitionId 'undefined': TypeError: partitionId is required. Given value: undefined. Hence it cannot be null or undefined.
    at Object.validateType (c:\sources\git\spew\SpewSocketIO\node_modules\@azure\event-processor-host\dist\lib\util\utils.js:24:15)
    at AzureStorageCheckpointLeaseManager.<anonymous> (c:\sources\git\spew\SpewSocketIO\node_modules\@azure\event-processor-host\dist\lib\azureStorageCheckpointLeaseManager.js:393:21)
    at Generator.next (<anonymous>)
    at c:\sources\git\spew\SpewSocketIO\node_modules\tslib\tslib.js:107:75
    at new Promise (<anonymous>)
    at Object.__awaiter (c:\sources\git\spew\SpewSocketIO\node_modules\tslib\tslib.js:103:16)
    at AzureStorageCheckpointLeaseManager.getCheckpoint (c:\sources\git\spew\SpewSocketIO\node_modules\@azure\event-processor-host\dist\lib\azureStorageCheckpointLeaseManager.js:392:24)
    at PartitionContext.<anonymous> (c:\sources\git\spew\SpewSocketIO\node_modules\@azure\event-processor-host\dist\lib\partitionContext.js:108:78)
    at Generator.next (<anonymous>)
    at c:\sources\git\spew\SpewSocketIO\node_modules\tslib\tslib.js:107:75. +0ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Calling _acquireExpiredInParallel with startAt 1, needed 0. +32ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Examining chunk at '3': [1], needed 0. +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Short circuit: needed is 0, unowned is: 4, off end -> true. +1ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Looking to steal: 0. +2ms
  azure:eph:partitionScanner [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] No need to scan further since remaining needed: 0. +1ms
  azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Did we steal any leases in this scan: false. +540ms
  azure:eph:partitionManager [js-host-496f4536-5d80-4d6c-8e1e-9c0b7c5029b2] Sleeping for 30 seconds before starting the next scan. +1ms
```

Reply:
@dan-drl -  C# uses pascal casing(PartitionId), whereas Node uses camel casing(partitionId). Therefore, this leads to undefined partitionId being set when we retrieve the blob content. 

 This fix is not a proper fix for this issue because your fix will only work for `partitionId`. Other properties like owner, token and offset still be undefined.

I'm closing this PR. I've created an issue here https://github.com/Azure/azure-sdk-for-js/issues/2508 to track down this.